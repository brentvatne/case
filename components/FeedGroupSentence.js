import React from 'react'
import { Text } from 'react-native'
import styled from 'styled-components/native'
import PropTypes from 'prop-types'
import UserNameText from './UserNameText'
import FeedWordLink from './FeedWordLink'
import { Typography, Units, Colors } from '../constants/Style'

const Container = styled.View`
  padding-vertical: ${Units.scale[2]};
  padding-horizontal: ${Units.scale[3]};
  margin-bottom: ${Units.base};
`

const Sentence = styled.Text`
  color: ${Colors.semantic.text};
  font-size: ${Typography.fontSize.medium};
  line-height: ${Typography.lineHeightFor('medium', 'compact')};
  font-weight: ${Typography.fontWeight.semiBold};
`

const Timestamp = styled.Text`
  margin-top: ${Units.scale[1]};
  color: ${Colors.gray.medium};
  font-size: ${Typography.fontSize.small};
  line-height: ${Typography.lineHeightFor('small', 'compact')};
`

const FeedGroupSentence = ({ group }) => {
  const { user, verb, connector, target, object } = group

  const targetColor = ((target && target.__typename === 'Channel')
    ? Colors.channel[target.visibility]
    : null
  )

  return (
    <Container>
      <Sentence>
        <UserNameText user={user} />

        <Text>
          {verb === 'commented' ? ': ' : ` ${verb} `}
        </Text>

        <FeedWordLink
          object={object}
          phrase={group.object_phrase}
        />

        <Text> {connector} </Text>

        <FeedWordLink
          object={target}
          phrase={group.target_phrase}
          style={{
            color: targetColor,
          }}
        />
      </Sentence>

      <Timestamp>
        {group.created_at}
      </Timestamp>
    </Container>
  )
}

FeedGroupSentence.propTypes = {
  group: PropTypes.any.isRequired,
}

export default FeedGroupSentence
