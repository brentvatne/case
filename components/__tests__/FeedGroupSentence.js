import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import FeedGroupSentence from '../FeedGroupSentence'

const group = {
  id: '3630_15271_added_1501702906',
  key: '3630_15271_added_1501702906',
  length: 1,
  user: {
    name: 'Ã‰douard U.',
    slug: 'edouard-u',
  },
  is_single: true,
  verb: 'connected',
  object: {
    __typename: 'Connectable',
    id: 1164746,
    title: 'tumblr_noj7p2hXzj1rz4sfeo1_r1_540.jpg',
  },
  object_phrase: 'an image',
  connector: 'to',
  target: {
    __typename: 'Channel',
    id: 15271,
    title: 'Picture',
    visibility: 'public',
  },
  target_phrase: 'Picture',
  created_at: '19 minutes ago',
}

test('renders correctly', () => {
  const tree = shallow(
    <FeedGroupSentence group={group} />,
  )
  expect(toJSON(tree)).toMatchSnapshot()
})
