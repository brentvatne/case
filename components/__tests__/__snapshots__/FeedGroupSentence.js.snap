// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Styled(View)>
  <Styled(Text)>
    <UserNameText
      onPress={[Function]}
      user={
        Object {
          "name": "Ã‰douard U.",
          "slug": "edouard-u",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
       
      connected
       
    </Text>
    <FeedWordLink
      object={
        Object {
          "__typename": "Connectable",
          "id": 1164746,
          "title": "tumblr_noj7p2hXzj1rz4sfeo1_r1_540.jpg",
        }
      }
      phrase="an image"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
       
      to
       
    </Text>
    <FeedWordLink
      object={
        Object {
          "__typename": "Channel",
          "id": 15271,
          "title": "Picture",
          "visibility": "public",
        }
      }
      phrase="Picture"
      style={
        Object {
          "color": "rgb(66, 133, 60)",
        }
      }
    />
  </Styled(Text)>
  <Styled(Text)>
    19 minutes ago
  </Styled(Text)>
</Styled(View)>
`;
